src/sentry/tasks/statistical_detectors.py

```diff
@@ -0,0 +1,135 @@
+            performance_projects.append(project.id)
+
+            if len(performance_projects) >= ITERATOR_CHUNK:
+                detect_regressed_transactions.delay(performance_projects)
+                performance_projects = []
+
+        if project.flags.has_profiles:
+            profiling_projects.append(project.id)
+
+            if len(profiling_projects) >= ITERATOR_CHUNK:
+                detect_regressed_functions.delay(profiling_projects, now)
+                profiling_projects = []
+    """
+
+    # make sure to dispatch a task to handle the remaining projects
+    if performance_projects:
+        detect_regressed_transactions.delay(performance_projects)
+        performance_projects = []
+    if profiling_projects:
+        detect_regressed_functions.delay(profiling_projects, now)
+        profiling_projects = []
+
+
+@instrumented_task(
+    name="sentry.tasks.statistical_detectors._detect_regressed_transactions",
+    queue="performance.statistical_detector",
+    max_retries=0,
+)
+def detect_regressed_transactions(project_ids: List[int], **kwargs) -> None:
+    if not options.get("statistical_detectors.enable"):
+        return
+
+    for project_id in project_ids:
+        query_transactions(project_id)
+
+
+@instrumented_task(
+    name="sentry.tasks.statistical_detectors._detect_regressed_functions",
+    queue="performance.statistical_detector",
+    max_retries=0,
+)
+def detect_regressed_functions(project_ids: List[int], start: datetime, **kwargs) -> None:
+    if not options.get("statistical_detectors.enable"):
+        return
+
+    for project in Project.objects.filter(id__in=project_ids):
+        try:
+            query_functions(project, start)
+        except Exception as e:
+            sentry_sdk.capture_exception(e)
+
+
+def query_transactions(project_id: int) -> None:
+    pass
+
+
+def query_functions(project: Project, start: datetime) -> None:
+    params = _get_function_query_params(project, start)
+
+    # TODOs:
+    # - format and return this for further processing
+    # - handle any errors
+    functions.query(
+        selected_columns=[
+            "timestamp",
+            "fingerprint",
+            "count()",
+            "p95()",
+        ],
+        query="is_application:1",
+        params=params,
+        orderby=["-count()"],
+        limit=100,
+        referrer=Referrer.API_PROFILING_FUNCTIONS_STATISTICAL_DETECTOR.value,
+        auto_aggregations=True,
+        use_aggregate_conditions=True,
+        transform_alias_to_input_format=True,
+    )
+
+
+def _get_function_query_params(project: Project, start: datetime) -> Dict[str, Any]:
+    # The functions dataset only supports 1 hour granularity.
+    # So we always look back at the last full hour that just elapsed.
+    start = start - timedelta(hours=1)
+    start = start.replace(minute=0, second=0, microsecond=0)
+
+    return {
+        "start": start,
+        "end": start + timedelta(minutes=1),
```

Comment:
I'm a little confused by comment above: should this timedelta be `hours=1`?

Reply:
I'll clarify the comment but the timestamps on the aggregate functions table are truncated to the hour so having `timedelta(minute=1)` already covers the whole range